### PR TITLE
Simplify

### PR DESCRIPTION
**Pull request by robyoder**
I started out wanting a way to get the locale that looked less hacky than `T._i18nInstance.locale`. It's pretty easy to just add a `locale()` method to `T` that returns that value. But as I looked at the code I realized the whole `I18n` class and instance weren't really necessary, and I could simplify a lot by removing them. So I did.

All of the tests pass. The only things that are not backward compatible are references to `setup` (now called `set`) , `_i18nInstance`, and some stricter TS types. But those are sufficient to bump the version to 0.6.0.

### CODE REVIEW

File: src/format.ts [hunk 1]
```diff
@@ -1,58 +1,56 @@
...
-		style: 'percent'
+		style: "percent"
 	}
 }
-numberFormatOptions.default = numberFormatOptions.decimal;
 
 // Create cached versions of Intl.DateTimeFormat and Intl.NumberFormat
-export default function createCachedFormatter(intlFormat: (IntlFormatType)): CachedFormatter {
-	let cache:any = {};
+export default function createCachedFormatter<T extends IntlFormat>(intlFormat: IntlFormatType<T>): CachedFormatter<T> {
```
Comment (mitchchn) [inline src/format.ts]: Please use X for generics.

File: src/format.ts [hunk 2]
```diff
@@ -1,58 +1,56 @@
...
+	let cache: any = {};
 
-	return function (locale: string, formatOptions?: IntlFormatOptions): IntlFormat {
+	return function (locale: string, formatOptions?: IntlFormatOptions<T>): T {
 		const args = Array.prototype.slice.call(arguments);
 		const id = locale + "-" + JSON.stringify(formatOptions);
 		if (id in cache) return cache[id];
 
-		const formatter: IntlFormat = new (Function.prototype.bind.call(
+		const formatter: T = new (Function.prototype.bind.call(
```
Comment (mitchchn) [inline src/format.ts]: Same here.

File: src/t-i18n.ts [hunk 3]
```diff
@@ -1,132 +1,103 @@
...
 /**
  * T-i18n - lightweight localization
- * v0.5.0
+ * v0.6.0
  *
  * T-i18n defers to standards to do the hard work of localization. The browser Intl API is use to format
  * dates and numbers. Messages are provided as functions rather than strings, so they can be compiled at build time.
- *
- *
- * Cobbled together by Mitch Cohen on July 31, 2017
```
Comment (mitchchn) [inline src/t-i18n.ts]: Instead of removing my name, add yours. :)

File: src/t-i18n.ts [hunk 4]
```diff
@@ -1,132 +1,103 @@
...
-//
 export interface TFunc {
 	(message: string, replacements?: IcuReplacements, id?: string): string;
+	generateId: (message: string) => string;
+	locale: () => string;
 	lookup: (id: string, replacements?: IcuReplacements, defaultMessage?:string) => string
 	setup: (options?: SetupOptions) => any;
-	date: (value: any, formatName?: string, locale?: string) => string;
-	number: (value: any, formatName?: string, locale?: string) => string;
+	date: (value: any, formatName?: keyof typeof dateTimeFormats, locale?: string) => string;
```
Comment (mitchchn) [inline src/t-i18n.ts]: Unless I'm mistaken, the `value` for both the date and number functions is `number`. They should be typed regardless.

BTW, it's outside the scope of this PR but I'd love to be able to pass in custom formatter options.

File: src/t-i18n.ts [hunk 5]
```diff
@@ -1,132 +1,103 @@
...
+	let locale = defaultLanguage;
+	let idGenerator = generator.hyphens;
 
-		if (messages && messages[key]) {
-			return messages[key];
-		}
-		if (defaultMessages && defaultMessages[key]) {
-			return defaultMessages[key];
-		}
+	const setup = (options: SetupOptions = {}): SetupOptions => {
```
Comment (mitchchn) [inline src/t-i18n.ts]: Would the `Partial` type be useful here?

File: src/format.ts [hunk 6]
```diff
@@ -1,58 +1,56 @@
...
+		style: "percent"
 	}
 }
-numberFormatOptions.default = numberFormatOptions.decimal;
 
 // Create cached versions of Intl.DateTimeFormat and Intl.NumberFormat
-export default function createCachedFormatter(intlFormat: (IntlFormatType)): CachedFormatter {
-	let cache:any = {};
+export default function createCachedFormatter<T extends IntlFormat>(intlFormat: IntlFormatType<T>): CachedFormatter<T> {
+	let cache: any = {};
```
Comment (mitchchn) [inline src/format.ts]: Can you type the cache?

File: src/format.ts (hunk 1 above)
Comment (robyoder) [reply src/format.ts]: Oh yeah I forgot. :)

File: src/t-i18n.ts (hunk 3 above)
Comment (robyoder) [reply src/t-i18n.ts]: Well it wasn't just your name. This project is no longer a one-day cobbling. :) But I can add names back.

**Pull request merged**